android/src/main/java/com/pagopa/ioreactnativecrypto/X509VerificationUtil.kt

```diff
@@ -0,0 +1,428 @@
+   *
+   * @property isValid True if the certificate chain is valid and trusted, false otherwise.
+   * @property status The detailed status code indicating the outcome or reason for failure.
+   * @property errorMessage A descriptive message, especially in case of failure.
+   * @property failingCertificate The certificate that caused the validation failure, if applicable.
+   */
+  data class ValidationResult(
+    val isValid: Boolean,
+    val status: ValidationStatus,
+    val errorMessage: String? = null,
+    val failingCertificate: X509Certificate? = null
+  )
+
+  /**
+   * Verifies a certificate chain against a trust anchor.
+   * Revocation checks using CRLs are performed *only if* CRL Distribution Points (CDPs)
+   * are specified in the certificates within the chain.
+   *
+   * @param certChainBase64 List of Base64 encoded certificates, starting with the end-entity cert and ending with the CA cert.
+   * @param trustAnchorCertBase64 Base64 encoded trust anchor (CA) certificate.
+   * @param options Configuration for network timeouts.
+   * @return A [ValidationResult] indicating the outcome.
+   */
+  suspend fun verifyCertificateChain(
+    certChainBase64: List<String>,
+    trustAnchorCertBase64: String,
+    options: X509VerificationOptions
+  ): ValidationResult {
+    if (certChainBase64.isEmpty()) {
+      return ValidationResult(false, ValidationStatus.INVALID_CHAIN_PATH, "Certificate chain is empty.")
+    }
+
+    val certificateFactory: CertificateFactory
+    val trustAnchorCert: X509Certificate
+    val certificateChain: List<X509Certificate>
+    val trustAnchor: TrustAnchor
+
+    // --- 1. Decode Certificates and Trust Anchor ---
+    try {
+      certificateFactory = CertificateFactory.getInstance(CERT_TYPE)
+      val trustAnchorBytes = Base64.decode(trustAnchorCertBase64, Base64.DEFAULT)
+      trustAnchorCert = certificateFactory.generateCertificate(ByteArrayInputStream(trustAnchorBytes)) as X509Certificate
+      trustAnchor = TrustAnchor(trustAnchorCert, null)
+
+      certificateChain = certChainBase64.map { certBase64 ->
+        val certBytes = Base64.decode(certBase64, Base64.DEFAULT)
+        certificateFactory.generateCertificate(ByteArrayInputStream(certBytes)) as X509Certificate
+      }
+
+      // Check if the last certificate in the chain was issued by the trust anchor
+      val lastCertInChain = certificateChain.lastOrNull()
+        ?: return ValidationResult(false, ValidationStatus.INVALID_CHAIN_PATH, "Certificate chain is effectively empty after decoding.")
+      // Allow the chain to end with the trust anchor itself OR a cert issued by the trust anchor.
+      if (!lastCertInChain.encoded.contentEquals(trustAnchorCert.encoded) &&
+        lastCertInChain.issuerX500Principal != trustAnchorCert.subjectX500Principal) {
+        return ValidationResult(false, ValidationStatus.INVALID_TRUST_ANCHOR, "Last certificate in chain not issued by trust anchor.")
+      }
```

Comment:
I would tighten this check. I would find the first certificate in the chain issued by the trust anchor and consider that as the "last certificate". There could be alterations to the chain where attackers could lengthen the chain.